backend/FwLite/FwLiteProjectSync.Tests/EntrySyncTests.cs

```diff
@@ -67,6 +66,156 @@ public Task DisposeAsync()
     private readonly SyncFixture _fixture = fixture;
     protected IMiniLcmApi Api = null!;
 
+    private static readonly AutoFaker AutoFaker = new(AutoFakerDefault.MakeConfig(
+        ExtraWritingSystemsSyncFixture.VernacularWritingSystems));
+
+    public enum ApiType
+    {
+        Crdt,
+        FwData
+    }
+
+    // Not all of these cases are realistic, but they should all work
+    // An especially critical case is syncing data to crdt that has been round-tripped through fwdata first.
+    [Theory]
+    [InlineData(ApiType.Crdt)]
+    [InlineData(ApiType.FwData)]
+    [InlineData(null)]
+    public async Task CanSyncRandomEntries(ApiType? roundTripApiType)
+    {
+        // arrange
+        var currentApiType = Api switch
+        {
+            FwDataMiniLcmApi => ApiType.FwData,
+            CrdtMiniLcmApi => ApiType.Crdt,
+            // This works now, because we're not currently wrapping Api,
+            // but if we ever do, then we want this to throw, so we know we need to detect the api differently.
+            _ => throw new InvalidOperationException("Unknown API type")
+        };
+
+        IMiniLcmApi? roundTripApi = roundTripApiType switch
+        {
+            ApiType.Crdt => _fixture.CrdtApi,
+            ApiType.FwData => _fixture.FwDataApi,
+            _ => null
+        };
+
+        var before = AutoFaker.Generate<Entry>();
+        var after = AutoFaker.Generate<Entry>();
+        after.Id = before.Id;
+
+        await Api.PrepareToCreateEntry(before);
+        await Api.PrepareToCreateEntry(after);
+
+        if (roundTripApi is not null && currentApiType != roundTripApiType)
```

Comment:
❓ I don't yet understand this test well enough to see: What's the value in continuing rather than returning early `if (roundTripApi is null)`?